test/io.jl

```diff
@@ -327,8 +327,19 @@ module TestIO
     io = IOBuffer(abnormal*",%_B*\tC*,end\n1,2,3\n")
     @test names(readtable(io)) == ns
 
-    # Test writetable with append
+    # Test writetable with NA and compare to the results
+    tf = tempname()
+    isfile(tf) && rm(tf)
+    df = DataFrame(A = @data([1,NA]), B = @data(["b", NA]))
+    writetable(tf, df)
+    @test readcsv(tf) == ["A" "B"; 1 "b"; "NA" "NA"]
 
+    # Test writetable with nastring set and compare to the results
+    isfile(tf) && rm(tf)
+    writetable(tf, df, nastring="NaN")
+    @test readcsv(tf) == ["A" "B"; 1 "b"; "NaN" "NaN"]
```

Comment:
Looks like readcsv handles 'NaN', so the matrix literal would have to be `[1 2; 3 "4"; NaN NaN]`, though testing the behavior of readcsv isn't great, so changing the nastring would probably be better.